apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/tour-launch.js

```diff
@@ -93,12 +87,22 @@ function WelcomeTour() {
 					} );
 				},
 				onMaximize: ( currentStepIndex ) => {
-					setJustMaximized( true );
 					recordTracksEvent( 'calypso_editor_wpcom_tour_maximize', {
 						is_gutenboarding: isGutenboarding,
 						slide_number: currentStepIndex + 1,
 					} );
 				},
+				onStepViewOnce: ( currentStepIndex ) => {
+					const lastStepIndex = tourSteps.length - 1;
```

Comment:
`tourSteps` is a bit of a side effect here as it's coming from the outer context (same with `isGutenboarding`). We may need to rethink these a bit, but not sure exactly how right now 🤔

Reply:
Good news is that `isGutenboarding` is already going to be removed in [60040](https://github.com/Automattic/wp-calypso/issues/60040). We need to think about what to do with `tourSteps` here, if there is another solution apart from passing used values to the `onStepViewOnce` function

Reply:
We can come back to this in a follow-up.